Project/Source/Components/UI/ComponentToggle.cpp

```diff
@@ -3,39 +3,132 @@
 #include "Utils/Logging.h"
 #include "GameObject.h"
 #include "Components/UI/ComponentEventSystem.h"
+#include "Components/UI/ComponentImage.h"
+#include "Components/ComponentBoundingBox2D.h"
 #include "Application.h"
 #include "Modules/ModuleUserInterface.h"
-
+#include "Modules/ModuleScene.h"
+#include "Modules/ModuleResources.h"
+#include "Resources/ResourceScript.h"
+#include "Scripting/Script.h"
+#include "imgui.h"
+#include "Utils/ImGuiUtils.h"
 #include "Utils/Leaks.h"
 
 #define JSON_TAG_IS_ON "IsOn"
+#define JSON_TAG_ENABLED_IMAGE_ID "EnabledImageID"
+#define JSON_TAG_CLICKED_COLOR "ClickedColor"
 
 ComponentToggle ::~ComponentToggle() {}
 
 void ComponentToggle::Init() {
+	OnValueChanged();
+}
+
+void ComponentToggle::OnClicked() {
+	SetOn(!IsOn());
+
+	App->userInterface->GetCurrentEventSystem()->SetSelected(GetOwner().GetComponent<ComponentSelectable>()->GetID());
+
+	for (ComponentScript& scriptComponent : GetOwner().GetComponents<ComponentScript>()) {
```

Comment:
I don't think a Checkbox should have this OnClick Button action. It should only change the boolean of the Component. We already a have button that has the purpose of calling the click function of the scripts.

Reply:
We do need a way for a callback to be called, establishing an OnToggleClicked seems too dirty for me. Removed it anyway

Reply:
I don't think a callback is needed because the Component should be used in an informative way. What I mean by informative is that, an external entity will check that the Component is checked, and if it is, is gonna perform an action, otherwise it would be a button that triggers the action